src/mono/wasm/Wasm.Build.Tests/AppSettingsTests.cs

```diff
@@ -21,22 +21,49 @@ public AppSettingsTests(ITestOutputHelper output, SharedBuildPerTestClassFixture
     {
     }
 
+    public static IEnumerable<object?[]> LoadAppSettingsBasedOnApplicationEnvironmentData()
+    {
+        // Defaults
+        yield return new object?[] { false, null, null, "Development" };
+        yield return new object?[] { true, null, null, "Production" };
+
+        // Override defaults from MSBuild
+        yield return new object?[] { false, "Production", null, "Production" };
+        yield return new object?[] { true, "Development", null, "Development" };
+
+        // Override defaults from JavaScript
+        yield return new object?[] { false, null, "Production", "Production" };
+        yield return new object?[] { true, null, "Development", "Development" };
+
+        // Override MSBuild from JavaScript
+        yield return new object?[] { false, "FromMSBuild", "Production", "Production" };
+        yield return new object?[] { true, "FromMSBuild", "Development", "Development" };
+    }
+
     [Theory]
-    [InlineData("Development")]
-    [InlineData("Production")]
-    public async Task LoadAppSettingsBasedOnApplicationEnvironment(string applicationEnvironment)
+    [MemberData(nameof(LoadAppSettingsBasedOnApplicationEnvironmentData))]
+    public async Task LoadAppSettingsBasedOnApplicationEnvironment(bool publish, string msBuildApplicationEnvironment, string queryApplicationEnvironment, string expectedApplicationEnvironment)
```

Comment:
in `#nullable enable` shouldn't we have 

```suggestion
    public async Task LoadAppSettingsBasedOnApplicationEnvironment(bool publish, string? msBuildApplicationEnvironment, string? queryApplicationEnvironment, string? expectedApplicationEnvironment)
```

Reply:
I'll push it once the CI succeeds